### ISSUE TITLE
Generating text with multiple different prefixes

### ISSUE THREAD

**Issue by hc09141** [minimaxir/gpt-2-simple#62]
I've fine-tuned a GPT2 model and am wanting to examine content it generates when given several different prefixes. Do you have any plans to allow generating with multiple prefixes within a single batch?

**Comment by minimaxir** [comment]
tl;dr no. I am doing work with [ray](https://github.com/minimaxir/tensorflow-multiprocess-ray) to allow asynchronous prediction which could theoretically be extended to the GPU, but that's way out of scope for this project.

**Comment by lkemkes** [comment]
@hc09141: I am in a similar situation. Have you found a workaround for this?

**Comment by hc09141** [comment]
@lkemkes I didn't have time to make a good workaround for this, but I came up with a slightly hacky solution. I found that starting the tensor flow session each time was taking quite a few seconds. However, I also found that after generating content a few times, it would take longer and longer to generate (probably due to the variables not being deleted in the tensorflow session after being used). My temporary solution was just to reset and restart the tensorflow session every n generations (I used 50), which ended up being feasible for the project I was working on.

Let me know if you find a better solution!
